### PR TITLE
Pydantic v2 compatibility

### PR DESCRIPTION
**Pull request by pjbull**

### CODE REVIEW

**Comment by pjbull** [comment]
Auto-running was disabled because of lack of repo activity(?), here's the test run:
https://github.com/drivendataorg/nbautoexport/actions/runs/5503874277

File: nbautoexport/sentinel.py [hunk 1]
```diff
@@ -50,6 +50,20 @@ class NbAutoexportConfig(BaseModel):
     class Config:
         extra = "forbid"
 
+    # deprecated in pydantic v2.0
```
Comment (jayqi) [inline nbautoexport/sentinel.py:53]: I don't think "deprecated" is correct. This should say "renamed".

**Review by jayqi — CHANGES_REQUESTED** [review]
Fix comments

File: nbautoexport/sentinel.py (hunk 1 above)
Comment (pjbull) [reply nbautoexport/sentinel.py:53]: They do still work, but have a deprecation warning:

```
PydanticDeprecatedSince20: The `dict` method is deprecated; use `model_dump` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.0.3/migration/
```

Comment (jayqi) [reply nbautoexport/sentinel.py:53]: Ah, you're right. That's my mistake.

**Review by jayqi — APPROVED** [review]

**Pull request merged**